src/lib/logical_query_plan/projection_node.cpp

```diff
@@ -29,6 +29,33 @@ bool ProjectionNode::is_column_nullable(const ColumnID column_id) const {
   return node_expressions[column_id]->is_nullable_on_lqp(*left_input());
 }
 
+std::shared_ptr<LQPUniqueConstraints> ProjectionNode::unique_constraints() const {
+  auto unique_constraints = std::make_shared<LQPUniqueConstraints>();
+  unique_constraints->reserve(node_expressions.size());
+
+  // Forward unique constraints, if applicable
+  const auto& input_unique_constraints = left_input()->unique_constraints();
+
+  const auto& expressions = this->output_expressions();
+  const auto output_expressions = ExpressionUnorderedSet{expressions.cbegin(), expressions.cend()};
```

Comment:
naming

Reply:
Weird. I wonder why no compiler complained about these expressions being unused?

Reply:
> Weird. I wonder why no compiler complained about these expressions being unused?

Me too. In other cases, it acts rather strict. For example:
```cpp
const auto [_, success] = column_expressions.emplace(column_expression);
DebugAssert(success, "Did not expect multiple column expressions for the same column id.");
```

The CI has failed because of `success` – which was unused in Release builds of course 😑

Fix became rather verbose:
```cpp
if constexpr (HYRISE_DEBUG) {
  const auto [_, success] = column_expressions.emplace(column_expression);
  Assert(success, "Did not expect multiple column expressions for the same column id.");
} else {
  column_expressions.emplace(column_expression);
}
```

Reply:
I think a regular assert would not have killed us there. After all, it is just a single branch.

Reply:
I already had a discussion with @mweisgut on this. Happy to discuss it on Wednesday.